core/src/main/java/com/scalar/db/api/TableMetadata.java

```diff
@@ -169,4 +169,44 @@ public TableMetadata build() {
           columns, partitionKeyNames, clusteringKeyNames, clusteringOrders, secondaryIndexNames);
     }
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
+    TableMetadata metadata = (TableMetadata) o;
+
+    if (!Objects.equals(columnNames, metadata.columnNames)) {
```

Comment:
Maybe, we can simplify this comparison using ComparisonChain of guava.
https://github.com/google/guava/wiki/CommonObjectUtilitiesExplained#comparecompareto

Reply:
I was not aware of this, It's a nice utility method! 
Though all the class attributes are objects that do not implement the `Comparable` interface I am not sure it's possible to use in this case
```
  private final LinkedHashSet<String> columnNames;
  private final Map<String, DataType> columnDataTypes;
  private final LinkedHashSet<String> partitionKeyNames;
  private final LinkedHashSet<String> clusteringKeyNames;
  private final Map<String, Order> clusteringOrders;
  private final Set<String> secondaryIndexNames;
  ```

Reply:
Or we can simply write as follows so that we can remove `return false` statements?
```
return (Objects.equals(columnNames, metadata.columnNames) &&
    Objects.equals(columnDataTypes, metadata.columnDataTypes) &&
    ...);
```

Reply:
Sure, let me update it.